flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/StateLocalitySlotAssigner.java

```diff
@@ -116,66 +116,53 @@ public Collection<SlotAssignment> assignSlots(
             JobInformation jobInformation,
             Collection<? extends SlotInfo> freeSlots,
             VertexParallelism vertexParallelism) {
-        Collection<? extends SlotInfo> remainingSlots = freeSlots;
-        final Collection<SlotAssignment> assignments = new ArrayList<>();
+        final List<ExecutionSlotSharingGroup> allGroups = new ArrayList<>();
         for (SlotSharingGroup slotSharingGroup : jobInformation.getSlotSharingGroups()) {
-
-            List<ExecutionSlotSharingGroup> sharedSlotToVertexAssignment =
-                    createExecutionSlotSharingGroups(vertexParallelism, slotSharingGroup);
-
-            SlotAssigner.AssignmentResult result =
-                    assignSlots(remainingSlots, sharedSlotToVertexAssignment);
-            remainingSlots = result.remainingSlots;
-            assignments.addAll(result.assignments);
+            allGroups.addAll(createExecutionSlotSharingGroups(vertexParallelism, slotSharingGroup));
         }
-        return assignments;
-    }
-
-    private AssignmentResult assignSlots(
```

Comment:
This method has now become rather long. There are some simple ways to reduce this, like moving all of this into a separate method:
```
scores = new PriorityQueue<>(Comparator.reverseOrder());
        for (ExecutionSlotSharingGroup group : allGroups) {
            calculateScore(group, parallelism)
                    .forEach(
                            (allocationId, score) ->
                                    scores.add(
                                            new AllocationScore(
                                                    group.getId(), allocationId, score)));
        }
```
same for
```
final Map<JobVertexID, Integer> parallelism = new HashMap<>();
        allGroups.forEach(
                group ->
                        group.getContainedExecutionVertices()
                                .forEach(
                                        evi ->
                                                parallelism.merge(
                                                        evi.getJobVertexId(), 1, Integer::sum)));
```

This is just boring setup work

Reply:
Addressed by d3595c472e38a96b66a9ee7ac1a394706802d431.